### PR TITLE
feature: фильтры и сортировки

### PR DESCRIPTION
**Pull request by Kupov**

### CODE REVIEW

**Comment by Kupov** [comment]
@dontuse

File: src/Filter/reducer.js [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  return prioritySort;
+};
+
+export default function app(state = initialState, action) {
+  switch (action.type) {
+    case CONFIG_LOAD_SUCCESS:
+      return {
+        ...state,
+        config: action.payload,
+      };
```
Comment (apress-hound) [inline src/Filter/reducer.js]: Unexpected '}'.

File: src/Filter/reducer.js [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+  return prioritySort;
+};
+
+export default function app(state = initialState, action) {
+  switch (action.type) {
+    case CONFIG_LOAD_SUCCESS:
+      return {
+        ...state,
+        config: action.payload,
```
Comment (apress-hound) [inline src/Filter/reducer.js]: Missing semicolon.<br>Expected 'case' and instead saw ':'.<br>Expected '}' to match '{' from line 31 and instead saw ':'.<br>Unrecoverable syntax error. (17% scanned).

File: src/Filter/reducer.js [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+  }
+
+  return prioritySort;
+};
+
+export default function app(state = initialState, action) {
+  switch (action.type) {
+    case CONFIG_LOAD_SUCCESS:
+      return {
+        ...state,
```
Comment (apress-hound) [inline src/Filter/reducer.js]: Expected '}' to match '{' from line 34 and instead saw '...'.<br>Missing semicolon.

File: src/Filter/reducer.js [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+  }
+
+  if (payload.id && !sorter.direction) {
+    return prioritySort + 1;
+  }
+
+  return prioritySort;
+};
+
+export default function app(state = initialState, action) {
```
Comment (apress-hound) [inline src/Filter/reducer.js]: Regular parameters should not come after default parameters.

File: src/App/sagas.js [hunk 5]
```diff
@@ -10,16 +10,16 @@ const getRubricatorData = () =>
...
   const tableData = yield call(getTableData);
-  yield put({type: 'LOAD_TABLE_DATA_SUCCESS', payload: tableData});
+  yield put({type: 'TABLE_EDITOR_LOAD_SUCCESS', payload: tableData});
   yield put({type: 'CLEAR_HISTORY', payload: {}});
 }
 
 export function* loadRubricatorData() {
   const trubricatorData = yield call(getRubricatorData);
-  yield put({type: 'LOAD_RUBRICATOR_DATA_SUCCESS', payload: trubricatorData.tree_nodes});
+  yield put({type: 'TREE_LOAD_SUCCESS', payload: trubricatorData.tree_nodes});
```
Comment (apress-hound) [inline src/App/sagas.js]: Identifier 'tree_nodes' is not in camel case.

File: src/ActionsPanel/e-actions-panel.scss [hunk 6]
```diff
@@ -37,12 +37,10 @@
...
   &-filter-box { padding-left: 20px; padding-right: 10px; border-left: 1px solid #e0e0e0; display: flex; width: 100%; align-items: center; }
   &-last-actions-box { padding-right: 10px; display: flex; align-items: center; padding-left: 30px; flex: 0 0 0; }
-  &-filter-item { white-space: nowrap; text-overflow: ellipsis; position: relative; display: block; }
-  &-filter-item-box { border-radius: 13px; background-color: #e0e0e0; padding: 0 40px 0 20px; height: 25px; line-height: 25px; transition: border-color 0.3s, background 0.3s; border: 1px solid #e0e0e0; display: block;
-  }
-  &-remove {
-    @include close-icon(#dc6263); content: ''; width: 12px; height: 12px; position: absolute; right: 15px; top: 7px; cursor: pointer;
-    &:hover + .e-actions-panel-filter-item-box { border-color: #dc6263; background: #fff; }
+  &-filter-item {
+    cursor: pointer; border-radius: 13px; background-color: #e0e0e0; padding: 0 40px 0 20px; height: 25px; line-height: 23px; transition: border-color 0.3s, background 0.3s; border: 1px solid #e0e0e0; white-space: nowrap; text-overflow: ellipsis; position: relative; display: block;
```
Comment (apress-hound) [inline src/ActionsPanel/e-actions-panel.scss]: `0.3` should be written without a leading zero as `.3`

File: dist/assets-table.bundle.js [hunk 7]
```diff
@@ -78,7 +78,7 @@ return /******/ (function(modules) { // webpackBootstrap
 /******/ 	}
 /******/ 	
 /******/ 	var hotApplyOnUpdate = true;
-/******/ 	var hotCurrentHash = "81519c9b8da8492a90fb"; // eslint-disable-line no-unused-vars
+/******/ 	var hotCurrentHash = "f9c683dfa78c37eafcbc"; // eslint-disable-line no-unused-vars
```
Comment (apress-hound) [inline dist/assets-table.bundle.js]: 'hotCurrentHash' was used before it was defined.<br>Mixed double and single quotes.

File: dist/assets-table.bundle.js [hunk 8]
```diff
@@ -3,14 +3,14 @@
...
 	else if(typeof define === 'function' && define.amd)
-		define(["react", "react-dom"], factory);
+		define(["react", "react-dom", "react-router"], factory);
 	else if(typeof exports === 'object')
-		exports["Assets-table"] = factory(require("react"), require("react-dom"));
+		exports["Assets-table"] = factory(require("react"), require("react-dom"), require("react-router"));
 	else
-		root["Assets-table"] = factory(root["React"], root["ReactDom"]);
-})(this, function(__WEBPACK_EXTERNAL_MODULE_2__, __WEBPACK_EXTERNAL_MODULE_7__) {
+		root["Assets-table"] = factory(root["React"], root["ReactDom"], root["ReactRouter"]);
```
Comment (apress-hound) [inline dist/assets-table.bundle.js]: Expected '{' and instead saw 'root'.<br>Mixed double and single quotes.

File: dist/assets-table.bundle.js [hunk 9]
```diff
@@ -3,14 +3,14 @@
...
 (function webpackUniversalModuleDefinition(root, factory) {
 	if(typeof exports === 'object' && typeof module === 'object')
-		module.exports = factory(require("react"), require("react-dom"));
+		module.exports = factory(require("react"), require("react-dom"), require("react-router"));
 	else if(typeof define === 'function' && define.amd)
-		define(["react", "react-dom"], factory);
+		define(["react", "react-dom", "react-router"], factory);
 	else if(typeof exports === 'object')
-		exports["Assets-table"] = factory(require("react"), require("react-dom"));
+		exports["Assets-table"] = factory(require("react"), require("react-dom"), require("react-router"));
```
Comment (apress-hound) [inline dist/assets-table.bundle.js]: Expected '{' and instead saw 'exports'.<br>Mixed double and single quotes.

File: dist/assets-table.bundle.js [hunk 10]
```diff
@@ -3,14 +3,14 @@
 /*! eslint-disable */
 (function webpackUniversalModuleDefinition(root, factory) {
 	if(typeof exports === 'object' && typeof module === 'object')
-		module.exports = factory(require("react"), require("react-dom"));
+		module.exports = factory(require("react"), require("react-dom"), require("react-router"));
 	else if(typeof define === 'function' && define.amd)
-		define(["react", "react-dom"], factory);
+		define(["react", "react-dom", "react-router"], factory);
```
Comment (apress-hound) [inline dist/assets-table.bundle.js]: Expected '{' and instead saw 'define'.<br>Mixed double and single quotes.

**Pull request merged**